tutorials/calibration_flow/gptq_like.py

```diff
@@ -0,0 +1,273 @@
+        for inp in args[1]:
+            act_obs2(inp)
+
+        input_scale1, input_zp1 = act_obs1.calculate_qparams()
+        input_scale2, input_zp2 = act_obs2.calculate_qparams()
+
+        # we can optimize/modify the module here
+        module.input_scale1 = input_scale1
+        module.input_zp1 = input_zp1
+        module.input_scale2 = input_scale2
+        module.input_zp2 = input_zp2
+
+        # rerun the module with quantized and dequantized inputs
+        new_input1 = []
+        for inp in args[0]:
+            new_input1.append(fake_quantize_affine(inp, inp.shape, input_scale1, input_zp1, torch.uint8))
+
+        new_input2 = []
+        for inp in args[1]:
+            new_input2.append(fake_quantize_affine(inp, inp.shape, input_scale2, input_zp2, torch.uint8))
+
+        mt1 = MultiTensor(new_input1)
+        mt2 = MultiTensor(new_input2)
+
+        output_with_fake_quantized_inputs = module(mt1, mt2)
+        # we can return the modified output so it can be consumed by future modules
+        return output_with_fake_quantized_inputs
+
+    def _register_forward_hook(module: torch.nn.Module):
+        forward_hook_handle = module.register_forward_hook(
+            forward_hook, with_kwargs=True
+        )
+        module._forward_hook_handle = forward_hook_handle
+        return module
+
+    _replace_with_custom_fn_if_matches_filter(
+        model, _register_forward_hook, _is_two_linear
+    )
+
+# using a function to align with the API in quant_api
+def apply_activation_static_quant():
+    def _apply_activation_static_quant(observed_two_linear):
+        target_dtype = torch.uint8
+
+        linear1 = observed_two_linear.linear1
+        # activation quantization
+        act_scale, act_zero_point = observed_two_linear.input_scale1, observed_two_linear.input_zp1
+        input_quant_func1 = lambda x: to_affine_quantized_static(x, act_scale, act_zero_point, x.shape, target_dtype)
+        linear1.weight = torch.nn.Parameter(to_linear_activation_quantized(linear1.weight, input_quant_func1), requires_grad=False)
+
+        linear2 = observed_two_linear.linear2
+        # activation quantization
+        act_scale, act_zero_point = observed_two_linear.input_scale2, observed_two_linear.input_zp2
+        input_quant_func2 = lambda x: to_affine_quantized_static(x, act_scale, act_zero_point, x.shape, target_dtype)
+        linear2.weight = torch.nn.Parameter(to_linear_activation_quantized(linear2.weight, input_quant_func2), requires_grad=False)
+        del observed_two_linear.input_scale1
+        del observed_two_linear.input_zp1
+        del observed_two_linear.input_scale2
+        del observed_two_linear.input_zp2
+        return observed_two_linear
+
+    return _apply_activation_static_quant
+
+
+example_inputs = (torch.randn(32, 64), torch.randn(32, 64),)
```

Comment:
Up until this point this PR read as fairly abstract, I'm not sure who the intended audience is

Is this someone trying to implement a gptq like workflow in AO? That's an audience of 1? Also how are multitensors and forward hooks relevant for gptq exactly seems like there's a missing step in the explanation

Reply:
this is any flow that's similar to GPTQ, that I described in the beginning, but I'll add more explanations/motivation.

current known use cases are GPTQ, AutoRound and SpinQuant as mentioned in the summary